specification/sections/05-events.md

```diff
@@ -197,4 +214,22 @@ see: [provider event types](../types.md#provider-events), [provider events](#51-
 
 The `PROVIDER_ERROR` is not emitted from the provider itself; the SDK implementation must run the `PROVIDER_ERROR` handlers if the `on context changed` throws or otherwise signals an error.
 
-see: [provider event types](../types.md#provider-events), [provider events](#51-provider-events), context, [provider context reconciliation](02-providers.md#26-provider-context-reconciliation)
\ No newline at end of file
+see: [provider event types](../types.md#provider-events), [provider events](#51-provider-events), context, [provider context reconciliation](02-providers.md#26-provider-context-reconciliation)
+
+#### Requirement 5.3.5
+
+> If the provider emits an event, the value of the client's `provider status` **MUST** be updated accordingly.
+
+Some providers may emit events spontaneously, based on changes in their internal state (connections, caches, etc).
+The SDK must update it's internal representation of the provider's state accordingly:
+
+| Event                            | Associated Status                                       |
+| -------------------------------- | ------------------------------------------------------- |
+| `PROVIDER_READY`                 | `READY`                                                 |
+| `PROVIDER_STALE`                 | `STALE`                                                 |
+| `PROVIDER_ERROR`                 | `ERROR`                                                 |
+| `PROVIDER_CONFIGURATION_CHANGED` | N/A (provider remains in state `READY`)                 |
```

Comment:
```suggestion
| `PROVIDER_CONFIGURATION_CHANGED` | N/A (provider remains in its current state)                 |
```

I think this address the non-ready states like `PROVIDER_READY` and `PROVIDER_STALE`.